### DISCUSSION TITLE
How does UncontrolledProviderScope works?

### DISCUSSION THREAD

**Discussion by Trung15010802** [rrousselGit/riverpod#3692] [Q&A] [answered]
I don't use `UncontrolledProviderScope ` and it works fine so why we need `UncontrolledProviderScope ` ?
![image](https://github.com/user-attachments/assets/7d567cec-fc53-453c-8f72-b2b91fae48b4)

**Comment by rrousselGit** [comment]
We don't "need" it. It's just an implementation detail that leaked and that folks started to use, so I'm not going to remove it.

**Reply by Trung15010802** [reply]
if I only want to use ref in main method then I just need to use `ProviderContainer` without `UncontrolledProviderScope ` right?

**Reply by rrousselGit** [reply, marked as answer]
The recommended approach isn't to put logic in the main. Refer to https://riverpod.dev/docs/essentials/eager_initialization
